ocs_ci/ocs/hsbench.py

```diff
@@ -0,0 +1,209 @@
+        self.end_point_port = self.hsbench_cr["end_point_port"]
+        self.hsbench_bin_dir = self.hsbench_cr["hsbench_bin_dir"]
+
+    def create_resource_hsbench(self):
+        """
+        Create resource for hsbench mark test:
+            Create service account
+            Create PVC
+            Create golang pod
+
+        """
+        # Check for existing rgw pods on cluster
+        self.rgw_pod = pod.get_rgw_pods()
+        if self.rgw_pod:
+            # Create service account
+            self.sa_name = helpers.create_serviceaccount(self.namespace)
+            self.sa_name = self.sa_name.name
+            helpers.add_scc_policy(sa_name=self.sa_name, namespace=self.namespace)
+
+            # Create test pvc+pod
+            log.info(f"Create Golang pod to generate S3 workload... {self.namespace}")
+            pvc_size = "50Gi"
+            node_name = "compute-0"
+            self.pod_name = "hsbench-pod"
+            self.pvc_obj = helpers.create_pvc(
+                sc_name=constants.DEFAULT_STORAGECLASS_RBD,
+                namespace=self.namespace,
+                size=pvc_size,
+            )
+            self.pod_obj = helpers.create_pod(
+                constants.CEPHBLOCKPOOL,
+                namespace=self.namespace,
+                pod_name=self.pod_name,
+                pvc_name=self.pvc_obj.name,
+                node_name=node_name,
+                sa_name=self.sa_name,
+                pod_dict_path=self.pod_dic_path,
+                dc_deployment=True,
+                deploy_pod_status=constants.STATUS_COMPLETED,
+            )
+        else:
+            raise UnexpectedBehaviour(
+                "This cluster doesn't have RGW pod(s) to perform hsbench"
+            )
+
+    def install_hsbench(self, timeout=2400):
+        """
+        Install HotSauce S3 benchmark:
+        https://github.com/markhpc/hsbench
+
+        """
+        # Install hsbench
+        log.info(f"Installing hsbench S3 benchmark on testing pod {self.pod_obj.name}")
+        hsbench_path = "github.com/markhpc/hsbench"
+        self.pod_obj.exec_cmd_on_pod(f"go get -v {hsbench_path}", timeout=timeout)
+        log.info("Successfully installing hsbench benchmark")
+
+    def create_test_user(self):
+        """
+        Create a radosgw test user for S3 access
+
+        """
+        # Create RWG test user
+        self.toolbox = pod.get_ceph_tools_pod()
+        self.uid = self.hsbench_cr["uid"]
+        display_name = self.hsbench_cr["display_name"]
+        email = self.hsbench_cr["email"]
+        self.access_key = self.hsbench_cr["access_key"]
+        self.secret_key = self.hsbench_cr["secret_key"]
+
+        log.info(f"Create RGW test user {self.uid}")
+        self.toolbox.exec_cmd_on_pod(
+            f"radosgw-admin user create --uid={self.uid} "
+            f"--display-name={display_name} --email={email} "
+            f"--access-key={self.access_key} --secret-key={self.secret_key}"
+        )
+        return
```

Comment:
not needed

Reply:
we need credentials to create radosgw test user.